vaccine_feed_ingest/runners/ca/los_angeles_gov/normalize.py

```diff
@@ -0,0 +1,171 @@
+#!/usr/bin/env python
+
+import datetime
+import json
+import logging
+import os
+import pathlib
+import re
+import sys
+from typing import List, Optional
+
+from vaccine_feed_ingest_schema import location as schema
+
+from vaccine_feed_ingest.utils.normalize import provider_id_from_name
+from vaccine_feed_ingest.utils.parse import location_id_from_name
+
+# Configure logger
+logging.basicConfig(
+    level=logging.INFO,
+    format="%(asctime)s %(levelname)s:%(name)s:%(message)s",
+    datefmt="%m/%d/%Y %H:%M:%S",
+)
+logger = logging.getLogger("ca/los_angeles_gov/normalize.py")
+
+
+def _get_address(site: dict) -> schema.Address:
+    city = None
+    zip = None
+
+    m = re.match(r"(.+),?\s+(CA[,.]?\s*)?(?:(\d{5})[,-]?)?", site["addr2"])
+    if m:
+        city = m.group(1)
+        zip = m.group(3)
+
+    # A few entries have the whole address in the "addr1" field
+    street1 = site["addr1"]
+    m = re.search(r", Los Angeles, CA\s*(\d{5})?", street1)
+    if m:
+        street1 = street1[: m.start()]
+        city = "Los Angeles"
+        zip = m.group(1)
+
+    return schema.Address(street1=street1, city=city, state="CA", zip=zip)
+
+
+def _get_location(site: dict) -> Optional[schema.LatLng]:
+    if site["lat"] and site["lon"]:
+        return schema.LatLng(latitude=site["lat"], longitude=site["lon"])
+
+    return None
+
+
+def _get_contacts(site: dict) -> Optional[List[schema.Contact]]:
+    contacts = []
+
+    if re.match(r"^https?://", site["link"]):
+        contacts.append(schema.Contact(contact_type="booking", website=site["link"]))
+
+    if len(contacts) > 0:
+        return contacts
+
+    return None
+
+
+def _get_inventory(site: dict) -> Optional[List[schema.Vaccine]]:
+    # This is just a guess.
+    vaccines = []
+
+    if "j" in site["vaccines"]:
+        vaccines.append(schema.Vaccine(vaccine="johnson_johnson_janssen"))
+
+    if "m" in site["vaccines"]:
+        vaccines.append(schema.Vaccine(vaccine="moderna"))
+
+    if "p" in site["vaccines"]:
+        vaccines.append(schema.Vaccine(vaccine="pfizer_biontech"))
+
+    if len(vaccines) > 0:
+        return vaccines
+
+    return None
+
+
+def _get_parent_organization(site: dict) -> Optional[schema.Organization]:
+    maybe_provider = provider_id_from_name(site["name"])
```

Comment:
Thank you for using the shared function! This means that you automatically benefitted from improvements like #332!